src/dbnode/x/xio/types.go

```diff
@@ -39,8 +39,12 @@ type BlockReader struct {
 }
 
 // ReadSegment reads the Segment, blocking until the segment is available or the deadline expires.
-func (b BlockReader) ReadSegment(ctx context.Context) (segment ts.Segment, err error) {
+func (b BlockReader) ReadSegment(ctx context.Context) (ts.Segment, error) {
```

Comment:
@wesleyk this was interesting race

Reply:
nice

Reply:
actually not following the race, when would segment/err not be set?

Reply:
so the race was the go routine _and_ the case Done could both set return vars without a lock, which are basically local vars. now there are explicit local vars for the go routine to set and the case Done returns new vars.